src/ontime/core/plot/anomaly_plot.py

```diff
@@ -0,0 +1,212 @@
+                )
+                .transform_filter(alt.datum.anomalies == 1)
+            )
+
+            chart += anomalies_chart
+
+        if contextual_anomalies is not None:
+            chart = AnomalyPlot._make_line_chart(data_df, chart, contextual_anomalies)
+
+        if collective_anomalies is not None:
+            chart = AnomalyPlot._make_area_chart(data_df, chart, collective_anomalies)
+
+        if seasonal_anomalies is not None:
+            chart = AnomalyPlot._make_line_chart(data_df, chart, seasonal_anomalies)
+
+        if cyclical_anomalies is not None:
+            chart = AnomalyPlot._make_line_chart(data_df, chart, cyclical_anomalies)
+
+        return chart
+
+    @staticmethod
+    def _make_line_chart(
+        data: pd.DataFrame, actual_chart: alt.Chart, anomalies: BinaryTimeSeries
+    ) -> alt.Chart:
+        """
+        Make a line chart with the given anomalies.
+
+        :param data: TimeSeries within the data that are plotted in the chart.
+        :param actual_chart: Chart that will be updated with the anomalies.
+        :param anomalies: BinaryTimeSeries of anomalies that will be plotted in red on the chart.
+
+        :return: Return an altair chart with the current time series and the anomalies drawn.
+        """
+        array_anomalies_df = AnomalyPlot.split_continuous_series(anomalies)
+        chart_total = actual_chart
+        for anomalies_df in array_anomalies_df:
+            anomalies_df["anomalies_y"] = (
+                anomalies_df["anomalies"] * data[AnomalyPlot._DATA_COLUMN_NAME]
+            )
+            chart = (
+                alt.Chart(anomalies_df.reset_index())
+                .mark_line(color="red", strokeWidth=2.5)
+                .encode(
+                    x="time:T",
+                    y="anomalies_y:Q",
+                )
+            )
+            chart_total += chart
+
+        return chart_total
+
+    @staticmethod
+    def _make_area_chart(
+        data_df: pd.DataFrame, actual_chart: alt.Chart, anomalies: BinaryTimeSeries
+    ) -> alt.Chart:
+        """
+        Adding background to the chart with the given anomalies.
+
+        :param data_df: TimeSeries within the data that are plotted in the chart.
+        :param actual_chart: Chart that will be updated with the anomalies.
+        :param anomalies: BinaryTimeSeries of anomalies that will be used to color the background in red on the chart.
+
+        :return: Return an altair chart with the current time series and the anomalies drawn.
+        """
+
+        y_max = data_df[AnomalyPlot._DATA_COLUMN_NAME].max()
+        y_min = data_df[AnomalyPlot._DATA_COLUMN_NAME].min()
+
+        delta = (y_max - y_min) * 0.1
+
+        array_anomalies_df = AnomalyPlot.split_continuous_series(anomalies)
+        chart_total = None
+        for anomalies_df in array_anomalies_df:
+            anomalies_df["max"] = y_max + delta
+            anomalies_df["min"] = y_min - delta
+
+            chart = (
```

Comment:
When I did run the notebook, the area did not fill up the top and bottom of the plot. This could probably be fixed here.
![Screenshot 2023-12-04 at 10 37 08](https://github.com/fredmontet/ontime/assets/4543916/3351f931-d40a-482f-a7d4-bd382a525aa3)